envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadataValidator.kt

```diff
@@ -55,4 +61,17 @@ class NodeMetadataValidator(
 
     private fun isAllowedToHaveAllServiceDependencies(metadata: NodeMetadata) = properties
         .outgoingPermissions.servicesAllowedToUseWildcard.contains(metadata.serviceName)
+
+    private fun validateConfigurationMode(metadata: NodeMetadata) {
+        if (!properties.configurationMode.ads && !properties.configurationMode.xds) {
+            throw ConfigurationModeNotSupportedException(metadata.serviceName, "Neither ADS nor XDS")
```

Comment:
Is this "if" really necessary? This case is a misconfiguration of EC. I think we have two better ways:
1. Validate properties when EC starts (more work to do)
2. Simply delete this "if". We will know, that something is wrong anyway: When we start EC we will see many "ads/xds is not supported by server" messages in logs (assuming we also add log.warn() as I wrote in another comment). For me it's perfectly acceptable solution.

This "if" gives us nothing IMO, because without it we will have the same behaviour, just with another message ("Neither ADS nor XDS" vs "ADS"/"XDS")